index.html

```diff
@@ -49,19 +49,19 @@ <h1 class="large l-align-center l-mb-mega">&lt;gaia-button&gt;</h1>
   <hr/>
 
   <section>
-    <gaia-button><i data-icon="back-light" aria-hidden="true"></i><span>Back</span></gaia-button>
-    <gaia-button><span>Forward</span><i data-icon="forward-light" aria-hidden="true"></i></gaia-button>
+    <gaia-button><span data-l10n-id="back" data-icon="back-light">Back</span></gaia-button>
+    <gaia-button><span data-l10n-id="forward" data-icon-end="forward-light">Forward</span></gaia-button>
   </section>
 
   <hr/>
 
   <section flex>
-    <gaia-button circular><i data-icon="camera"></i></gaia-button>
-    <gaia-button circular><i data-icon="bug"></i></gaia-button>
-    <gaia-button circular><i data-icon="video"></i></gaia-button>
-    <gaia-button circular><i data-icon="camera"></i></gaia-button>
-    <gaia-button circular><i data-icon="reload"></i></gaia-button>
-    <gaia-button circular><i data-icon="airplane"></i></gaia-button>
+    <gaia-button circular><span data-l10n-id="camera" data-icon="camera"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="bug" data-icon="bug"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="video" data-icon="video"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="camera" data-icon="camera"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="reload" data-icon="reload"/></gaia-button>
+    <gaia-button circular><span data-l10n-id="airplane" data-icon="airplane"/></gaia-button>
```

Comment:
@wilsonpage - my comment in the thread was about loading l10n resources within the component (and storing them there). I said that we have no patterns or infra for that yet. So we should always store l10n resources in the client code and load it into the component via data-l10n-id.

```<gaia-button data-l10n-id="l10nId"></button>```

is fine.